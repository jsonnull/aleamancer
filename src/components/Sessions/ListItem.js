// @flow
import React from 'react'
import styled from 'styled-components'
import type { SessionInfo } from 'types'
import { fonts, fontSize } from 'styles/common'

const Tag = styled.div`
  display: inline-block;
  border-radius: 2px;
  height: 1.5rem;
  line-height: 1.5rem;
  font-size: ${fontSize.small};
  font-weight: bold;
  color: ${props => props.theme.backgroundSecondary};
`

const SessionName = styled.div`
  font-family: ${fonts.heading};
  font-size: ${fontSize.medium};
`

const Session = styled.div`
  padding: 1rem;
  border-radius: 5px;
  margin: 0 1rem 2rem;
  flex: 1 0 25%;
  max-width: 25%;
  min-height: 100px;
  cursor: pointer;
  background-color: ${props => props.theme.backgroundSecondary};
`

const Meta = (props: { isCurrent: boolean }) => {
  if (props.isCurrent) {
    return <Tag>Current</Tag>
  }

  return null
}

const Loading = () => (
  <div>
    <i className={`fa fa-circle-o-notch fa-spin fa-fw`} />{' '}
    <span>Retrieving game info...</span>
  </div>
)

type Props = {
  isCurrent: boolean,
  session: SessionInfo,
  setSession: Function
}

const Item = (props: Props) => {
  // FIXME: Add visible distinction of current game
  // const { isCurrent } = props

  const content = !props.session.meta ? (
    <Loading />
  ) : (
    <div>
      <SessionName>{props.session.meta.name}</SessionName>
      <Meta isCurrent={props.isCurrent} />
    </div>
  )

  return (
    <Session
      onClick={() => {
        props.setSession(props.session.id)
      }}
    >
      {content}
    </Session>
  )
}

export default Item
