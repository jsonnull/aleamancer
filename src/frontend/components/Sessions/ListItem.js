// @flow
import React from 'react'
import styled from 'styled-components'
import { fontSize, fonts } from 'frontend/styles/common'

const SessionName = styled.div`
  font-family: ${fonts.heading};
  font-size: ${fontSize.medium};
`

const Session = styled.div`
  padding: 1rem;
  border-radius: 5px;
  margin: 0 1rem 2rem;
  flex: 1 0 25%;
  max-width: 25%;
  min-height: 100px;
  cursor: pointer;
  background-color: ${props => props.theme.backgroundSecondary};
`

type Props = {
  name: string,
  setSession: Function
}

const Item = (props: Props) => {
  return (
    <Session onClick={props.setSession}>
      <SessionName>{props.name}</SessionName>
    </Session>
  )
}

export default Item
