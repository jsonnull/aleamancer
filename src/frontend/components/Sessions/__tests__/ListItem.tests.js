// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import ListItem from '../ListItem.js'

describe('Sessions ListItem component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ListItem name="name" setSession={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
