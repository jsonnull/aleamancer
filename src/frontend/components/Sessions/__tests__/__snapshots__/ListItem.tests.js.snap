// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sessions ListItem component renders correctly 1`] = `
.c2 {
  display: inline-block;
  border-radius: 2px;
  height: 1.5rem;
  line-height: 1.5rem;
  font-size: 1.3rem;
  font-weight: bold;
}

.c1 {
  font-family: Lora;
  font-size: 1.8rem;
}

.c0 {
  padding: 1rem;
  border-radius: 5px;
  margin: 0 1rem 2rem;
  -webkit-flex: 1 0 25%;
  -ms-flex: 1 0 25%;
  flex: 1 0 25%;
  max-width: 25%;
  min-height: 100px;
  cursor: pointer;
}

<div
  className="c0"
  onClick={[Function]}
>
  <div>
    <div
      className="c1"
    >
      testName
    </div>
    <div
      className="c2"
    >
      Current
    </div>
  </div>
</div>
`;
